chia/consensus/blockchain.py

```diff
@@ -917,44 +925,24 @@ async def get_block_generator(
                 reorg_chain[prev.height] = prev
                 curr = prev
 
-            peak: Optional[BlockRecord] = self.get_peak()
-            if self.contains_block(curr.prev_header_hash) and peak is not None:
-                # Then we look up blocks up to fork point one at a time, backtracking
-                previous_block_hash = curr.prev_header_hash
-                prev_block_record = await self.block_store.get_block_record(previous_block_hash)
-                prev_block = await self.block_store.get_full_block(previous_block_hash)
-                assert prev_block is not None
-                assert prev_block_record is not None
-                fork = find_fork_point_in_chain(self, peak, prev_block_record)
-                curr_2: Optional[FullBlock] = prev_block
-                assert curr_2 is not None and isinstance(curr_2, FullBlock)
-                reorg_chain[curr_2.height] = curr_2
-                while curr_2.height > fork and curr_2.height > 0:
-                    curr_2 = await self.block_store.get_full_block(curr_2.prev_header_hash)
-                    assert curr_2 is not None
-                    reorg_chain[curr_2.height] = curr_2
```

Comment:
we still need, at least, the height-to-hash mapping that `reorg_chain` provides. I don't think you can remove this whole seciont

Reply:
ok i get what wrong here.

we traverse all the heights in the end anyway and fetch the missing blocks from db but we do that by height so might get the wrong one, so i just need the mapping to get the hashes this code still redundant, we dont need all the full blocks up to the fork we only need the referenced ones

Reply:
by `block_records`, you mean the `Blockchain` object, right? The `self`. It has access to all blocks, including orphaned blocks and other forks. If we're in a reorg, those blocks are in the `Blockchain` object as well.

I think the intention of that if-statement, `if self.contains.block(curr.prev_header_hash)` was to ask if we know of that block, not if we happen to have it in the cache right now. If we assume the intention was the latter, there's another block of code missing here to do the same thing but pull the block hashes from the database.

the `additional_blocks` is just the current *batch* of block's we're validating. It does not contain the whole fork. The fork is still only stored in the database at this point, that's where we need to get the generators from.

Reply:
then i dont get the point of this check, im pretty sure the initial intention was to see if its in the chain

otherwise we are just checking the store to contain something that we expect to be there, in that case i would have expected an assert or some exception being thrown, what is the case then when the block not in blockchain.blockrecords ?